api/v1alpha1/vaultdynamicsecret_types.go

```diff
@@ -18,10 +18,25 @@ type VaultDynamicSecretSpec struct {
 	Namespace string `json:"namespace,omitempty"`
 	// Mount path of the secret's engine in Vault.
 	Mount string `json:"mount"`
+	// RequestHTTPMethod to use when syncing Secrets from Vault.
+	// Setting a value here is not typically required.
+	// If left unset the Operator will make requests using the GET method.
+	// In the case where Params are specified the Operator will use the PUT method.
+	// Please consult https://developer.hashicorp.com/vault/docs/secrets if you are
+	// uncertain about what method to use.
+	// +kubebuilder:validation:Enum={GET,PUT}
```

Comment:
PUT will work as well, but in all the API docs I've checked I've only found POST for generating creds (e.g. AWS STS and k8s). Are there examples of PUT as well? I'd say POST should definitely be a valid option, and optionally include PUT as a valid option if you have any counter examples.

Reply:
Funny the client API uses `PUT` 🤔

Reply:
lol 🤦 both PUT and POST just get translated into `logical.WriteOperation` in Vault, so it shouldn't make any difference, it's purely a documentation exercise AFAICT

Reply:
Fixed in 53b3b367ef1a78cb9d978916f545fc943cd081e6